doc/bum_storm_control/bum_storm_control_hld.md

```diff
@@ -0,0 +1,399 @@
+The storm-control feature allows the user to limit the amount of BUM traffic admitted to the system. This can be achieved by configuring the type of storm (Broadcast or Unknown-unicast or unknown-Multicast) and the corresponding kilo bits per second (kbps) parameter on a given physical interface. Traffic that exceeds the configured rate will be dropped. 
+Unknown-multicast traffic consists of all multicast traffic which donot match any of the statically configured or dynamically learned multicast groups. 
+
+
+
+## 1.1 Requirements
+
+
+### 1.1.1 Functional Requirements
+ 1. Support configuration of Broadcast, Unknown-unicast and unknown-Multicast storm-control independently on physical interfaces.
+ 2. Support threshold rate configuration in kilo bits per second (kbps) in the range of 0 kbps to 100,000,000 kbps (100Gbps). 
+
+### 1.1.2 Configuration and Management Requirements
+This feature supports Click, Klish, REST, gNMI interfaces.
+ 1. Support a CLI to add or delete broadcast, unknown-unicast and unknown-multicast storm-control on a physical interface as described in "Configuration Commands" section below. 
+ 2. Support show commands to display the storm-control configuration as described in "Show Commands" section below. 
+ 3. Support debug commands as described in "Debug Commands" section below.
+
+
+### 1.1.3 Scalability Requirements
+Storm-control configuration is done on physical ports. Hence it should be supported on all physical ports of the system. 
+
+### 1.1.4 Warm Boot Requirements
+Storm-control functionality should continue to work across warm reboot. 
+
+- To support planned system warm boot.
+- To support SWSS docker warm boot.
+
+
+## 1.2 Design Overview
+
+
+### 1.2.1 Basic Approach
+The Policer Orchestration agent is used for handling BUM storm-control configuration. 
+
+### 1.2.2 Container
+No new containers are introduced. Functionality is added to the existing swss container. 
+
+### 1.2.3 SAI Overview
+The existing SAI policer and interface APIs shall be used to support this feature and they are available at below location, 
+https://github.com/opencomputeproject/SAI/blob/master/inc/saipolicer.h <br>
+https://github.com/opencomputeproject/SAI/blob/master/inc/saiport.h
+
+
+The details of the changes will be discussed in the design section below.
+
+
+# 2 Functionality
+Refer to section 1
+## 2.1 Target Deployment Use Cases
+Any Layer-2 network segment where unknown-unicast / broadcast / unknown-multicast traffic storm should be  controlled. 
+
+
+## 2.2 Functional Description
+Refer to section 1.1
+
+## 2.3 Limitations
+BUM storm control 
+- Configuration is not supported on VLAN and port-channel interfaces. User can configure on physical port which is part of a VLAN / port-channel.
+- Statistics is not supported. 
+- User is expected to remove all storm-control configurations from interface before doing a breakout from Click CLI. 
```

Comment:
I think, we do force delete configs during breakout..right?

Reply:
Yes. will remove this comment. breakout to be handled in code.